app/views/books/_books_search.html.erb

```diff
@@ -0,0 +1,23 @@
+<%= form_tag(books_path, method: :get, id: 'search_form') do %>
+  <%= text_field_tag :search, params[:search], placeholder: "Search by book or by author" %>
+  <%= submit_tag "Search", name: nil %>
+<% end %>
+
+<%= link_to t('books.search.reset'), books_path %>
+
+<% if params[:search] %>
```

Comment:
I think there is a bug here where you may have a search with no results. If that's the case it will show this section of code and the code on line 21. So I think this should be something like `params[:search] && books.any?`

Reply:
I thought of this too, but I think that the way its currently written is a happy medium. Allows nice re-use of the existing listing view, for example.